airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -168,6 +168,7 @@ class KubernetesPodOperator(BaseOperator):
     :param labels: labels to apply to the Pod. (templated)
     :param startup_timeout_seconds: timeout in seconds to startup the pod.
     :param get_logs: get the stdout of the container as logs of the tasks.
+    :param logs_timeout: timeout in seconds to read logs after container termination.
```

Comment:
I am not sure this needs to be configurable.

It reads like a setting "if no logs for logs_timeout settings then quit". but... really this timeout is applied only after pod completes. and, in this case, 2 minutes seems fine, it's a potentially confusing param, and i don't think we need to make that configurable.

Reply:
Well, the problem occurs when we read logs after a container is being terminated. We researched this issue but the actual root cause wasn't found because it seems that the problem is at a very low level (third-party library `urllib3` hangs at this line: https://github.com/urllib3/urllib3/blob/d393b4a5091c27d2e158074f81feb264c5c175af/src/urllib3/response.py#L999).
Experiments show that reading logs works fine for about  2-3 minutes after container termination, but if not all logs were red then it hangs. Because we didn't figure out what exactly causes hanging, we decided to make this parameter configurable for cases if `urllib3`'s behavior will change in the future.

Reply:
If the container is done, shouldn't there be no more logs forthcoming? In which case, I'm not sure what need there would be to increase time beyond that. If there is a need to add configurability, we can always do so later. It's easy to add things like this but much more difficult to remove or rename. Current name is ambiguous.

Reply:
To be clear, I don't veto making it configurable. I recommend you don't make it configurable. And if you must make it configurable, I would suggest a more specific name.

Reply:
@dstandish, thank you for noting that. Actually, I agree that exposing this parameter isn't a good idea - it handles logic that doesn't belong to the operator's level of abstraction and we will have to maintain it. At the same time I'm sure that it's better to leave the ability for configuring it because updates of third-party libraries can change the system's behavior and users would be able to adjust this parameter and stop suffering.
As a compromise, could we consider a different approach based on envs or maybe something else? I didn't find any recommendations regarding envs in the contribution manuals.

Reply:
Well, I guess you could make it a constant on the consumer classes, then that could be monkey patched.  Though that's a more advanced / hacky way. But maybe sufficient?

Reply:
I think a constant is enough for this case. Really don't think it should be in the user facing api.

Reply:
done